plugin/pkg/kubectl/utils/utils.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclientset "k8s.io/client-go/kubernetes"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+)
+
+// NewClient uses the KUBECONFIG environment variable to create a new client
+// based on an existing configuration
+func NewClient() (*clientset.Clientset, *restclient.Config) {
+	// resolve kubeconfig location, prioritizing the --config global flag,
+	// then the value of the KUBECONFIG env var (if any), and defaulting
+	// to ~/.kube/config as a last resort.
+	home := os.Getenv("HOME")
```

Comment:
Does this work on windows? I remember Docker having to go thru a ton of hoops to get this info across various platforms. Additionally, we should check to make sure we can use "/" instead of os.PathSeparator

Reply:
check out https://github.com/mitchellh/go-homedir. my head is full of other things right now so I don't recall if it works on windows, but it avoids cgo and that's a start